### PR TITLE
[FEATURE] inline label text feature

### LINKED ISSUE #602: [FEATURE] - Allow floating bookmark label or annotation in the editor
Firstly, I really appreciate the extension.
Recently I am going to read the understand the source code of ceph, the project is huge and hard to understand, I find Bookmark extension is very useful, but lack some funtion that i think may be awesome.
I think it would be awesome if the bookmark label message can be rendered just in the editor like a floating window or else. Side Bar is used for navicating, and the floating label message will be useful to read the source code of a big project.

### PR DESCRIPTION
**Pull request by lioshik**
Hello. First of all, I would like to say thank you for creating such a useful extension, which I am happy to use.

While I was using the extension, I lacked the opportunity to see bookmark labels inside text editor (similar to how ErrorLens extension displays errors/warnings inline). This feature would be useful for me because I use labeled bookmarks from your extension as a better alternative to temporal comments with some notes and TODOs about code I am working on right now. I think other people will find this feature useful too.

I've noticed that similar feature request has been open for almost three years (https://github.com/alefragnani/vscode-bookmarks/issues/602) so I decided to implement this feature myself.

This pull request includes not just code, but also updates to README with example image and explanations how to set up inline label text appearance. I also added translations to other languages for all newly added setting options

I am open to any feedback and up to resolve issues in this pull request

### CODE REVIEW

**Comment by lioshik** [comment]
@alefragnani

**Review by alefragnani — CHANGES_REQUESTED** [review]
Hi @lioshik ,

First of all, thank you for your PR. 

I made a few appointments, so whenever possible, take a look. I still didn't have a chance to properly test it, but once I do, I'll let you know. But, on a wider view, it seems just fine.

Thank you

**Comment by lioshik** [comment]
I have fixed what was said in your comments

File: package.json [hunk 1]
```diff
@@ -691,6 +691,34 @@
...
+                        "oblique"
+                    ],
+                    "description": "%bookmarks.configuration.label.inline.fontStyle.description%"
+                },
+                "bookmarks.label.inline.fontWeight": {
+                    "type": "integer",
+                    "minimum": 100,
+                    "maximum": 900,
+                    "default": 450,
+                    "description": "%bookmarks.configuration.bookmarks.label.inline.fontWeight.description%"
```
Comment (alefragnani) [inline package.json]: I guess the duplicate `bookmarks` in the id was just a typo. Simplify to  `bookmarks.configuration.label.inline.fontWeight.description`

**Review by alefragnani — CHANGES_REQUESTED** [review]

**Review by alefragnani — COMMENTED** [review]
## Code Review

This is a well-structured feature PR that uses VS Code's `after` text decoration to show bookmark labels inline (similar to ErrorLens/git blame). The implementation approach is sound. Here are the issues I found:

---

### ❌ High Priority

**1. Missing Dutch (`nl`) localization updates**

`package.nls.nl.json` exists in the repo but is not updated with the four new `bookmarks.label.inline.*` setting keys and the two `bookmarks.colors.labelInline*` keys. Dutch users will see raw `%...%` placeholder text in the Settings UI.

Similarly, `walkthrough/defineLabelsForYourBookmarks.nls.nl.md` exists and is not updated with the new inline labels section (all other walkthrough localizations were updated).

**2. Documentation inconsistency — "inlay hints" vs "git blame"**

The walkthrough `.md` files (all languages) describe the text color default as _"same color as for **inlay hints** is used"_, but `package.json` and `package.nls.json` correctly say _"same color as for **git blame** text decoration is used"_, and `package.json` defaults correctly use `git.blame.editorDecorationForeground`. The walkthrough files need to be corrected to say "git blame" not "inlay hints".

---

### ⚠️ Medium Priority

**3. Image filename typos**

Both image files referenced across the PR have two typos in the name `bookmarks-with-lables-arrrowed.png`:
- `lables` → `labels`
- `arrrowed` → `arrowed`

This affects `README.md` and all walkthrough `.md` files.

**4. Non-standard `fontWeight` default of `450`**

Standard CSS `font-weight` values are multiples of 100. While 450 is technically valid CSS, it is unusual and may render inconsistently across fonts. The conventional values are `400` (normal) or `500` (medium). Consider defaulting to `400`.

---

### ℹ️ Low Priority

**5. Missing newline at end of `decoration.ts`**

The diff shows `\ No newline at end of file` for `src/decoration/decoration.ts`. All source files should end with a newline.

---

### Summary

| Priority | Issue |
|---|---|
| ❌ | Missing `package.nls.nl.json` new keys |
| ❌ | Missing `walkthrough/defineLabelsForYourBookmarks.nls.nl.md` update |
| ❌ | Walkthrough files say "inlay hints" instead of "git blame" for color default |
| ⚠️ | Image filename typos (`lables` and `arrrowed`) |
| ⚠️ | `fontWeight` default `450` is non-standard; prefer `400` |
| ℹ️ | Missing newline at end of `decoration.ts` |

The core implementation logic in `decoration.ts` and `extension.ts` is correct — proper cleanup on config change, inline decoration built using `after` render options with per-range `contentText`, and re-rendering when inline settings change.

**Comment by alefragnani** [comment]
Hi @lioshik ,

The _Missing Dutch (nl) localization updates_ probably has to do with rebasing, since new PRs where merged recently.

And BTW, I tested the recent changes and it seems to be working just fine. Congratulations, and thank you!

**Comment by lioshik** [comment]
Hi @alefragnani

Fixed all new issues

**Comment by alefragnani** [comment]
Hi @lioshik ,

I guess you forgot to update the 3 NLS key names I commented (remove the extra `bookmarks` word). Please review it.

* `bookmarks.configuration.bookmarks.label.inline.enabled.description` should be `bookmarks.configuration.label.inline.enabled.description`
* `bookmarks.configuration.bookmarks.label.inline.margin.description` should be `bookmarks.configuration.label.inline.margin.description`
* `bookmarks.configuration.bookmarks.label.inline.fontWeight.description` should be `bookmarks.configuration.label.inline.fontWeight.description`

Thank you

File: package.json [hunk 2]
```diff
@@ -691,6 +691,34 @@
                         "%bookmarks.configuration.confirmClear.enumDescriptions.commandPalette%",
                         "%bookmarks.configuration.confirmClear.enumDescriptions.sideBar%"
                     ]
+                },
+                "bookmarks.label.inline.enabled": {
+                    "type": "boolean",
+                    "default": false,
+                    "description": "%bookmarks.configuration.bookmarks.label.inline.enabled.description%"
```
[the remark this answers is not in the document]
Comment (lioshik) [reply package.json]: fixed

File: package.json [hunk 3]
```diff
@@ -691,6 +691,34 @@
...
+                },
+                "bookmarks.label.inline.margin": {
+                    "type": "integer",
+                    "minimum": 0,
+                    "default": 2,
+                    "description": "%bookmarks.configuration.bookmarks.label.inline.margin.description%"
```
[the remark this answers is not in the document]
Comment (lioshik) [reply package.json]: fixed

File: package.json (hunk 1 above)
Comment (lioshik) [reply package.json]: fixed

**Comment by lioshik** [comment]
Renamed these translation keys

**Review by alefragnani — APPROVED** [review]
LGTM! 🚀  

I could run the final tests and it seems to be working just fine. 

I found a small issue in an edge case, but don't worry, I will be fixing it myself later, since I already had plans to touch the corresponding areas.

Again, thank you for your PR 🤝
